NotificationService/NotificationService.BusinessLibrary/Providers/MSGraphNotificationProvider.cs

```diff
@@ -431,13 +438,24 @@ private async Task ProcessMeetingEntitiesIndividually(string applicationName, IL
                     }
                     else
                     {
-                        item.Status = item.TryCount <= this.maxTryCount ? NotificationItemStatus.Retrying : NotificationItemStatus.Failed;
+                        if (item.TryCount <= this.maxTryCount && (result.StatusCode == HttpStatusCode.TooManyRequests || result.StatusCode == HttpStatusCode.RequestTimeout))
+                        {
+                            item.Status = NotificationItemStatus.Retrying;
+                            _ = this.LogMailBoxExhaustedTelemetry(result.Result, item.NotificationId, item.EmailAccountUsed, false, traceProps);
```

Comment:
Same comment here, and why it is coming multiple times?

Reply:
it is multiple times because there are three methods,
processEntitiesInBatch, processNotificationIndividually, ProcessMeetingEntitiesIndividually